subprojects/core/src/main/java/org/gradle/execution/plan/DefaultNodeValidator.java

```diff
@@ -51,6 +71,18 @@ public boolean hasValidationProblems(LocalTaskNode node) {
                     .withUserManual(userManualReference.getId(), userManualReference.getSection())
                     .nagUser();
             });
-        return !problems.isEmpty();
+    }
+
+    private void reportErrors(List<TypeValidationProblem> problems, TaskInternal task, WorkValidationContext validationContext) {
+        ImmutableSet<String> uniqueErrors = problems.stream()
+                .filter(problem -> !problem.getSeverity().isWarning())
+                .map(TypeValidationProblemRenderer::renderMinimalInformationAbout)
+                .collect(ImmutableSet.toImmutableSet());
+        if (!uniqueErrors.isEmpty()) {
+            throw WorkValidationException.forProblems(uniqueErrors)
```

Comment:
Can you show me what the log for a build looks like when we fail it here? Does it still say something that the task failed or is the message somewhat different?